File: 백준/Bronze/6131. 완전 제곱수/완전 제곱수.py
'''
1 <= B, A <= 500
1 <= N <= 1000
'''
N = int(input())

cnt = 0
for B in range(1, 501):
    for A in range(B, 501):
        if B ** 2 + N == A ** 2:
            cnt += 1
print(cnt)

File: 백준/Bronze/6131. 완전 제곱수/완전 제곱수.cc
#include <bits/stdc++.h>

using namespace std;

// 1 <= B <= A <= 500
// 1 <= N <= 1000
int main() {
  int n;
  int cnt = 0;
  cin >> n;

  for (int a = 1; a <= 500; a++) {
    for (int b = a; b <= 500; b++) {
      if (b * b == a * a + n)
        cnt++;
    }
  }
  cout << cnt << '\n';
}